src/rank_llm/retrieve_and_rerank.py

```diff
@@ -122,4 +145,4 @@ def retrieve_and_rerank(
                 for r in rerank_results
             ]
 
-    return rerank_results
+    return rerank_results[0]

```

Comment:
why? this will break the batch mode.
If you want to return the first element in intractive mode, do it with the condition